test/Chainweb/Test/RestAPI.hs

```diff
@@ -137,6 +137,18 @@ simpleClientSession envIO cid =
             (Expected gbh0)
             (Actual gen0)
 
+        void $ liftIO $ step "headerClient: get genesis block header pretty"
+        gen01 <- headerClientJsonPretty version cid (key gbh0)
+        assertExpectation "header client returned wrong entry"
+            (Expected gbh0)
+            (Actual gen01)
+
+        void $ liftIO $ step "headerClient: get genesis block header binary"
+        gen02 <- headerClientJsonBinary version cid (key gbh0)
+        assertExpectation "header client returned wrong entry"
+            (Expected gbh0)
+            (Actual gen02)
+
```

Comment:
Add more, and more targeted testing for different content types. So, far `tcpdump` was used to verify the content format on the wire. This should be done in Haskell code.

Reply:
Manual inspection of the content format revealed a messed up accept header for the default JSON encoding:

```
C00:20:55.546590 IP localhost.63240 > localhost.63238: Flags [P.], seq 4151126462:4151126669, ack 3782069362, win 6379, options [nop,nop,TS val 1907169050 ecr 1907169049], length 207
E.....@.@................m)..m.r...........
q...q...GET /chainweb/0.0/test-singleton/chain/0/header/CK3HRAiRYDfP21YK7La140gCwGOsFi9OwGeA2nomrf4 HTTP/1.1
Host: localhost:63238
Accept-Encoding: gzip
Accept: application/json;charset=utf-8,application/json


00:20:55.547049 IP localhost.63238 > localhost.63240: Flags [P.], seq 1:436, ack 207, win 6376, options [nop,nop,TS val 1907169051 ecr 1907169050], length 435
E.....@.@................m.r.m*............
q...q...HTTP/1.1 200 OK
Transfer-Encoding: chunked
Date: Mon, 16 Sep 2019 07:20:55 GMT
Server: Warp/3.3.1
Content-Type: application/json;charset=utf-8

0110
"AAAAAAAAAAAAAAAAAAAAAGGfHQl5Yv0fKhDv0lJixV3xhUPioCGQPNHoOFqFq_KhAAD__________________________________________92VICiaZyObrWKU6FiVjneIw0MhAdo358-uGfPwyNEoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGAAAAAAAAAAAAIrcdECJFgN8_bVgrstrXjSALAY6wWL07AZ4Daeiat_g"
0
```

For the other encodings the accept header was correct.

Reply:
Add test cases for all functions.